### PR TITLE
Move to go 1.21

### PR DESCRIPTION
**Pull request by rporres**
and update all dependencies

Since we're using ubi images, this is as far as we can go with go version.

One test needed to be adapted. The change in the prometheus client is to be blamed.

### CODE REVIEW

**Comment by rporres** [comment]
@janboll can you take a quick look into the test that needed to be changed. Maybe something has changed in the internal representation.

File: pkg/cache_test.go [hunk 1]
```diff
@@ -14,9 +14,9 @@ func createTestMetric(fqdn string, value float64) prometheus.Metric {
 }
 
 func TestGetMetricHash(t *testing.T) {
-	assert.Equal(t, "e2a622398e4a7ec5fca21510cc706fe28c1b37da32de620f0620804492890f62", getMetricHash(createTestMetric("foo_bar", 1)))
-	assert.Equal(t, "e2a622398e4a7ec5fca21510cc706fe28c1b37da32de620f0620804492890f62", getMetricHash(createTestMetric("foo_bar", 10)))
-	assert.NotEqual(t, "e2a622398e4a7ec5fca21510cc706fe28c1b37da32de620f0620804492890f62", getMetricHash(createTestMetric("other", 1)))
+	assert.Equal(t, "5e5435705ad2e07a1f989a92f230e6437dec1a12ae4f43fd26f74bcf8fa029cf", getMetricHash(createTestMetric("foo_bar", 1)))
```
Comment (janboll) [inline pkg/cache_test.go:17]: Odd, why did that change :thinking:

**Review by janboll — APPROVED** [review]

File: pkg/cache_test.go (hunk 1 above)
Comment (janboll) [reply pkg/cache_test.go:17]: Guess it's fine, the test looks like it covers this

Comment (rporres) [reply pkg/cache_test.go:17]: I really don't know. We have changed the version of the Prom golang client, but those structs (Counter, Metric) have not. At least superficially... 🤷

**Pull request merged**